### DISCUSSION TITLE
无限长序列、惰性计算与C++(1) – ChloroBlog

### DISCUSSION THREAD

**Discussion by Chlorie** [Chlorie/ChloroBlog#23] [Announcements]
https://chlorie.github.io/ChloroBlog/posts/2019-08-10/range-1.html 

 能鸽善鹉者的领域

**Comment by c0nstexpr** [comment]
```C++
using Intellichloire;
```

**Comment by Kenton1989** [comment]
无限循环太屑了（

在函数里定义struct太屑了（

不好好用class封装数据太屑了（

sentinel() != iterator()无定义太屑了（

**第六段代码把struct iterator打成struct iota_t太屑了（认真）**

屑盐酸（暴论）

**Comment by Chlorie** [comment]
> @Kenton1989
> 无限循环太屑了（
> 
> 在函数里定义struct太屑了（
> 
> 不好好用class封装数据太屑了（
> 
> sentinel() != iterator()无定义太屑了（
> 
> **第六段代码把struct iterator打成struct iota_t太屑了（认真）**
> 
> 屑盐酸（暴论）
> 

草
local struct主要是为了掩蔽名字，不然还得搞个`namespace impl`出来，不好展示。
反过来的不等不用定义，因为纯属为了实现范围`for`的功能。如果要实现一个正经的random access或者bidirection iterator的话我不如直接去世
boilerplate大放送
打错名字这个我错乐（
这两个关于lazy range的blog确实很屑，写的时候也没太过脑子，不如当黑历史（（

**Comment by Kenton1989** [comment]
> @Chlorie
> 草
> local struct主要是为了掩蔽名字，不然还得搞个`namespace impl`出来，不好展示。
> 反过来的不等不用定义，因为纯属为了实现范围`for`的功能。如果要实现一个正经的random access或者bidirection iterator的话我不如直接去世
> boilerplate大放送
> 打错名字这个我错乐（
> 这两个关于lazy range的blog确实很屑，写的时候也没太过脑子，不如当黑历史（（

大家都是学c艹的，为什么写得这么屑咱都懂，不用这么详细解释的（小声）
本来只想吐槽你那个typo，但是太闲了就一并吐槽了（小小声）
